### PR TITLE
"Added artillery, pools, and db routes.

### PR DESCRIPTION
**Pull request by sharmoo**
Removed legacy code.
Added new relic monitoring to test perofmance.
Created full crud functionality for comments and songs"

### CODE REVIEW

File: Artillery/perf.txt [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    min: 0.9
+    max: 2212.8
+    median: 55.1
+    p95: 457.2
+    p99: 1173.6
+  Scenario counts:
+    Song CRUD: 13563 (100%)
+  Codes:
+    200: 94941
+    201: 13563
```
Comment (pafrias) [inline Artillery/perf.txt]: I am of the opinion that this entire file should be git ignored

File: Artillery/config.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            comment: "Second."
+            commenttime: 0
+            username: "sharmoo"
+      - delete:
+            url: "/song/{{ songId }}/api/song_comment"
+            json:
+              id: "{{ commentId }}"
+
+      - delete:
+            url: "/song/{{ songId }}/api/song_id"
```
Comment (pafrias) [inline Artillery/config.yml:68]: What songId or range of songIds does this resolve to when testing?

**Review by pafrias — CHANGES_REQUESTED** [review]
Why are there two index.js files that are both server files?

File: server/db.js [hunk 3]
```diff
@@ -87,7 +87,7 @@ const updateSong = (songdata) => {
   const { id, songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl } = songdata;
 
   return new Promise((resolve, reject) => {
-    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8) where id = $1';
+    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8, $9) where id = $1';
```
Comment (pafrias) [inline server/db.js]: Could you explain the syntax here? In particular, I'd like to know about what lead you to define a function inside the Promise and then invoke it within the promise.

Bear in mind, I'm not entirely familiar with the client.release() function or using 'try'

File: server/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+config.js
+newrelic.js
```
Comment (pafrias) [inline server/.gitignore]: Good job 😜

File: server/index.js [hunk 5]
```diff
@@ -109,7 +79,7 @@ app.route('/song/:songId/api/song_id')
 app.route('/song/:songId/api/song_comment')
   //get all comments
   .get((req, res) => {
-    console.log(req.params.songId)
+    //console.log(req.params.songId)
```
Comment (pafrias) [inline server/index.js:82]: Is this the real index.js?

File: Artillery/perf.txt (hunk 1 above)
Comment (sharmoo) [reply Artillery/perf.txt]: Okay. Done

File: server/db.js (hunk 3 above)
Comment (pafrias) [reply server/db.js]: This may help clarify my curiosity:
https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns

File: server/db.js [hunk 6]
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
...
+    })
+    .catch(e => {
+      client.release()
+      return e.stack;
+    });
+};
+
 const createSong = (songdata) => {
   // destructure the songdata into its fields
   const { songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl } = songdata;
```
Comment (pafrias) [inline server/db.js:40]: very nice use of object deconstruction

File: server/db.js [hunk 7]
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
...
-        reject(e.stack);
-      });
-  });
-};
 
 
+const readASong = (songid) => {
+  const query = 'SELECT * from song where id = $1';
+  const params = [songid];
+  return pool.connect()
```
Comment (pafrias) [inline server/db.js:24]: Much cleaner!

File: server/db.js [hunk 8]
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
...
-    }
-    updateIt()
-      .catch(e => {
-        reject(e.stack)
-      })
-  })
+  const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8, $9) where id = $1';
+  return pool.connect()
+    .then(client => {
+      client.query(query, [id, songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl]);
```
Comment (pafrias) [inline server/db.js:81]: should this be returned?

File: server/newrelic.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+'use strict'
```
Comment (pafrias) [inline server/newrelic.js:1]: ... oh no....

Comment (sharmoo) [reply server/newrelic.js:1]: I'm laughing. And crying. HOW

Comment (sharmoo) [reply server/newrelic.js:1]: server/newrelic.js vs newrelic.js

**Pull request closed without merging**